ec2/networkinterface.py

```diff
@@ -0,0 +1,213 @@
+# #######
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+'''
+    ec2.NetworkInterface
+    ~~~~~~~~~~~~~~~~~~~~
+    AWS EC2 Network Interface
+'''
+
+
+# Cloudify imports
+from ec2 import connection
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+import boto.exception
+
+
+NODE_PROPERTIES_CREATE = [
+    'subnet_id', 'private_ip_address',
+    'description', 'groups'
+]
+
+SUBNET_RELATIONSHIP = \
+    'cloudify.aws.relationships.eni_contained_in_subnet'
+SUBNET_TYPE = 'cloudify.aws.nodes.Subnet'
+
+SECURITY_GROUP_RELATIONSHIP = \
+    'cloudify.aws.relationships.eni_connected_to_security_group'
+SECURITY_GROUP_TYPE = 'cloudify.aws.nodes.SecurityGroup'
+
+
+@operation
+def create(**_):
```

Comment:
is there a scenario where we would want the params as input to the operation here